social_simulator.py

```diff
@@ -4,36 +4,47 @@
 from random import *
 
 a=Tk()
-canvas=Canvas(a, width=400, height=300, bg='black')
+canvas=Canvas(a, width=500, height=400, bg='black')
 canvas.pack()
 canvas.focus_set()
 
 state = False
 kropki=[]
-
+b=time()
 
 def take_coords(ev):
     global kropki
     kropki += [Kropka([ev.x, ev.y])]
     for e in kropki:
-        e.f = choice(kropki[:-1]) if len(kropki)>1 else None
-        e.enemy = choice(kropki[:-1]) if len(kropki)>1 else None
+        while e == e.friend or e.friend == None:
+            e.friend = choice(kropki) if len(kropki)>1 else None
+        while e == e.enemy or e.enemy == None:
+            e.enemy = choice(kropki) if len(kropki)>1 else None
 
 def switch(ev):
     global state
     state = not state
+
+def debug(ev):
+    if len(kropki)>1:
+        for e in kropki:
+            print('<kropka nr '+str(kropki.index(e))+' przyjaciel: '+str(kropki.index(e.friend))+', wróg: '+str(kropki.index(e.enemy)))
+        print('\n')

```

Comment:
This prints two `'\n'`.
`print()` has the default `end='\n'` kwarg so behavior of `print()` = `print('\n',end='')`